### PR TITLE
Add fluentd.metrics.overrideOutputConf

### LINKED ISSUE #1022: Add an option to override output config for fluentd metrics
We should add an option to override the output config for fluentd metrics similar to how we provide `overrideOutputConf` for the logs section.

### PR DESCRIPTION
**Pull request by sumo-drosiek**
###### Description

Closes #1022 

###### Testing performed

- [ ] ci/build.sh
- [ ] Redeploy fluentd and fluentd-events pods
- [ ] Confirm events, logs, and metrics are coming in

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Do we need that? Customer can overwrite metrics configuration with `fluentd.metrics.rawConfig`

**Comment by vsinghal13** [comment]
> @vsinghal13 Do we need that? Customer can overwrite metrics configuration with `fluentd.metrics.rawConfig`

I created this issue to have feature parity with how we expose this for logs. We have `overrideOutputConf` and `rawConfig` for logs as well.

**Review by vsinghal13 — APPROVED** [review]

File: deploy/helm/sumologic/conf/metrics/metrics.conf [hunk 1]
```diff
@@ -68,4 +71,5 @@
 {{- end -}}
 {{ include "utils.metrics.match" (dict "Context" $ctx "Drop" $source.drop "Tag" $source.tag "Id" $source.id "Endpoint" (include "terraform.sources.name" (dict "Name" $endpoint "Type" "metrics")) "Storage" (index $ctx.Values.fluentd.buffer.filePaths.metrics $name) ) | nindent 4 }}
 {{- end }}
+{{- end }}
 </label>
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/conf/metrics/metrics.conf:75]:
```suggestion
</label>

```

**Comment by sumo-drosiek** [comment]
Close and reopen to re-generate checksum for overrides

**Review by frankreno — APPROVED** [review]
LGTM

**Review by perk-sumo — COMMENTED** [review]
👍

**Pull request merged**